src/Core/Block.cpp

```diff
@@ -397,6 +402,18 @@ size_t Block::allocatedBytes() const
     return res;
 }
 
+size_t Block::allocatedMetadataBytes() const
+{
+    size_t res = 0;
+    for (const auto & elem : data)
+        res += elem.column->allocatedMetadataBytes();
+
+    res += sizeof(data) + data.capacity() * sizeof(ColumnWithTypeAndName);
```

Comment:
ColumnWithTypeAndName itself will allocate, we are counting short ?

Reply:
yes, but it's difficult to collect them.